docs/okeps/okep-50088-evpn.md

```diff
@@ -0,0 +1,422 @@
+address reachability as well as IP address consistency. With MAC-VRFs and EVPN it becomes possible to extend the
+layer 2 network between the kubernetes cluster and outside world:
+
+![](images/l2-evpn-vm-migration.png)
+
+The image above depicts a Layer 2 UDN which not only exists across the worker nodes node-1 and node-2 but is also stretched
+into Provider Network 1. In this scenario, vm-2 is able to migrate into node-1 on the UDN network, preserving the same IP
+address it had in the external provider network. Similarly, there is another Provider Network 2 which may or may not
+correspond to another UDN within the Kubernetes cluster. However, notice that the red and blue networks are both using
+the same IP addressing scheme and sharing the same hardware, however due to VRF isolation they are completely unaware and
+unable to communicate with each other.
+
+### Using EVPN as the Overlay for Tenant Networks
+
+With integrating into a customer's already exising TOR spine and leaf architecture, Geneve can be disabled, and network
+segmentation will still persist for east/west traffic due to VXLAN tunnels with EVPN. This is true for both IP-VRFs,
+and  MAC-VRFs. This reduces packet overhead for customers, while also providing some other advantages that come with
+EVPN, such as link redundancy and broadcast, unknown unicast, and multicast (BUM) traffic suppression.
+
+## Proposed Solution
+
+EVPN will continue to build upon the BGP support already implemented into OVN-Kubernetes using FRR. This support includes
+integration with an OVN-Kubernetes API as well as an FRR-K8S API for configuring BGP peering and routing for UDNs. FRR
+already supports EVPN and similar API resources will be leveraged to accomplish configuring FRR as the BGP/EVPN control
+plane. A new EVPN CRD will be introduced as the API necessary to enable EVPN for a cluster and selected UDNs.
+
+FRR's implementation relies on specific Linux constructs in order to use EVPN:
+
+1. A VRF device
+2. Linux Bridge enslaved to the VRF
+3. An SVI attached to the bridge (in most cases)
+4. A VXLAN device enslaved to the bridge
+5. A VTEP IP configured locally (generally on a loopback interface)
+
+These objects are required to be configured to interoperate with FRR EVPN, and therefore this enhancement proposes a new
+controller functionality in OVN-Kubernetes that will be used to manage objects 2-5. The VRF itself is already managed by
+OVN-Kubernetes as part of the UDN itself. Furthermore, due to the reliance of FRR on these Linux devices, there is no current
+way to natively integrate the data path of EVPN into OVN/OVS. This may change in the future, but for now all traffic
+needs to traverse the Linux Kernel Networking stack. Due to this constraint, this enhancement proposes EVPN support only
+for local gateway mode.
+
+### Workflow Description
+
+Tenants as well as admins are able to create UDNs for their namespace, however it requires admin access to be able to
```

Comment:
Is this `UDNs` referring to CUDN or common UDN, or both?

Reply:
CUDN must be created by an admin user - hence I *suspect* this refers to UDNs (namespace scoped).

The fact is a namespace owner can create a UDN for their namespaces, but only an admin can expose said UDN via EVPN. Or that's how I interpret this.

Reply:
yeah, you are right to call this out. I should specifically say only CUDNs can be advertised by BGP and integrate with EVPN for the time being.